sql/api/src/main/scala/org/apache/spark/sql/catalyst/util/QuotingUtils.scala

```diff
@@ -71,4 +71,17 @@ object QuotingUtils {
 
     builder.toString()
   }
+
+  def escapeSpecialChar(str: String): String = {
```

Comment:
shall we put this function in `V2ExpressionSQLBuilder`? Maybe some dialects want to override it as they have more special chars.